src/Core/Maintenance/SalesChannel/Command/SalesChannelUpdateUrlCommand.php

```diff
@@ -0,0 +1,139 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Maintenance\SalesChannel\Command;
+
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\System\SalesChannel\Aggregate\SalesChannelDomain\SalesChannelDomainCollection;
+use Shopware\Core\System\SalesChannel\Aggregate\SalesChannelDomain\SalesChannelDomainEntity;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Constraints\NotEqualTo;
+use Symfony\Component\Validator\Constraints\Url;
+use Symfony\Component\Validator\Validation;
+
+/**
+ * @internal
+ */
+#[AsCommand(
+    name: 'sales-channel:update:url',
+    description: 'Updates a sales channel with a new URL',
+)]
+#[Package('discovery')]
+class SalesChannelUpdateUrlCommand extends Command
```

Comment:
IMO we don’t need a completely new command for this use-case.
The existing `Shopware\Core\Maintenance\SalesChannel\Command\SalesChannelUpdateDomainCommand` is internal anyway and could simply be refactored.

A more generic command that supports both scenarios via clear options would be easier to use and maintain.

For example:
`bin/console sales-channel:update --previous-domain=localhost --new-url=https://www.shopware.com/shop`

We could also introduce `--previous-url` so the lookup can be done via full URLs as well.

Reply:
@MartinKrzykawski Thank you, I have committed the changes in this commit https://github.com/shopware/shopware/pull/13282/commits/6722cc68f751c6837a73fbab9283f666ea387209